### PR TITLE
Show/hide (extension) task list and move completed tasks to bottom

### LINKED ISSUE #6638: Show/hide task list and move completed tasks to bottom
This is an update to the extended task list component on the home screen.

- As tasks are completed, they're moved to the bottom of the list.
- There’s a popover menu accessible from the card menu with the option to show/hide task list. The task list can be shown again from the Display menu in the top bar.
- Completed tasks can be deleted from the task ellipses menu (no confirmation needed)

Screen Shot 2021-04-01 at 4 37 01 PM

Screen Shot 2021-04-01 at 4 35 50 PM

Screen Shot 2021-04-01 at 4 31 56 PM

Tracks events:
`wcadmin_extended_tasklist_view` 
property: task_name
When user views the Extended task list on the home screen
`wcadmin_extended_tasklist_hide`
When user hides the Extended task list from the card menu
`wcadmin_extended_tasklist_show`
When user shows the Extended task list from the Display dropdown

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #6638.

This PR sorts the extension task list by completion status and allows for toggling the list visibility via the display options menu.

Additionally, the list title has been changed to "Things to do next" and the "view" tracking event name has been corrected (there was a duplicate event name for each list).

In a potentially controversial move 🌶️  I added a SlotFill to the Display Options menu in order to implement the conditional visibility of the "show things to do next" item. (See: https://github.com/woocommerce/woocommerce-admin/issues/6638#issuecomment-819574780) I think SlotFill is a cleaner approach than connecting the `<DisplayOptions>` component to everything that determines what tasks are shown.

### Accessibility

-   [x] I've tested using only a keyboard (no mouse)
-   [ ] I've tested using a screen reader
-   [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
-   [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

Screen Shot 2021-04-13 at 9 52 30 AM
_Completed tasks at the bottom_

Screen Shot 2021-04-13 at 9 52 51 AM
_Visibility toggle_

### Detailed test instructions:

- Ensure you have multiple extension tasks (use this diff if you'd like: https://gist.github.com/jeffstieler/32cf7e591173e9d17c47f0a6ee2ac7e6)
- Select the "Example 1" task and mark it complete
- Verify that it is sorted to the bottom of the task list
- Click "display options"
- Toggle the "things to do next" list off
- Verify the list is no longer visible
- Verify the `wcadmin_extended_tasklist_hide` event is fired

### CODE REVIEW

**Review by moon0326 — COMMENTED** [review]
Nice work @jeffstieler  👍 Code looks clean and the test worked as expected with `add-task` plugin.

I'm not sure if it's intended, but I got `Cannot read property 'sort' of undefined` without `add-task` plugin enabled. I left a comment on the line.

File: client/task-list/index.js [hunk 1]
```diff
@@ -124,9 +124,17 @@ export class TaskDashboard extends Component {
 		} = this.props;
 		const { isCartModalOpen } = this.state;
 		const allTasks = this.getAllTasks();
-		const { extension: extensionTasks, setup: setupTasks } = allTasks;
+		const { extension, setup: setupTasks } = allTasks;
 		const { task } = query;
 
+		const extensionTasks = extension.sort( ( a, b ) => {
```
Comment (moon0326) [inline client/task-list/index.js:130]: I'm not sure if it's intended, but I get `Cannot read property 'sort' of undefined` from here without `add-task` enabled.

Comment (jeffstieler) [reply client/task-list/index.js:130]: Strange, I'm not having that error occur locally, but I can reproduce it by modifying the example to return a non-array. 🤷‍♂️

Comment (jeffstieler) [reply client/task-list/index.js:130]: Oh whoops - just needed to test without the example plugin activated!

Comment (jeffstieler) [reply client/task-list/index.js]: Fixed in a45344d.

**Comment by jeffstieler** [comment]
@moon0326 this is ready for another review. Please note that there has been a significant change - the addition of SlotFill to implement this feedback: https://github.com/woocommerce/woocommerce-admin/issues/6638#issuecomment-819574780.

_cc: @samueljseay_

**Comment by moon0326** [comment]
> @moon0326 this is ready for another review. Please note that there has been a significant change - the addition of SlotFill to implement this feedback: [#6638 (comment)](https://github.com/woocommerce/woocommerce-admin/issues/6638#issuecomment-819574780).
> 
> _cc: @samueljseay_

Thank you for the change. I will review it first thing in the morning (sorry I saw the change a bit late today) 👍

**Review by moon0326 — APPROVED** [review]
The new slot code changes look good!!! 👍 x 1000 and tested well 🚢 🚢 

Thank you for the Slot code change :) I was looking for a good example of it.

**Comment by jeffstieler** [comment]
> Thank you for the Slot code change :) I was looking for a good example of it.

🙇‍♂️ 

This is a simple example since the Fill isn't publicly consumable (for now) and doesn't restrict or transform the child components.

**Comment by moon0326** [comment]
> > Thank you for the Slot code change :) I was looking for a good example of it.
> 
> 🙇‍♂️
> 
> This is a simple example since the Fill isn't publicly consumable (for now) and doesn't restrict or transform the child components.

Still though :) I can start with it. I think..the example on the official doc is a bit confusing tbh.

**Review by samueljseay — APPROVED** [review]
As discussed on other channels, I'm happy with the SlotFill, it keeps everything together so even if we didn't like the approach later its not a drama to change it. 🚀

**Pull request merged**